condition_resource_contains_test.go

```diff
@@ -26,6 +26,15 @@ func TestResourceContains(t *testing.T) {
 		value     string
 		pass      bool
 	}{
+		//Correct matching within resource string without delimiter:
+		{matches: "Exact resource component match in resource string, with no delimiter", value: ":GroupId:2:", resource: "resources:sensor-data:GroupId:2:deviceIds:a9b576e8-7419-4eed-a010-7f68ec0ff588", pass: true},
+		{matches: "Exact resource component match in resource string, with no delimiter", value: "myResource", resource: "myResource", pass: true},
+		//Better to use regex for string, to avoid false positives:
+		{matches: "Exact resource component match in resource string, with no delimiter", value: "(notMyResource$)", resource: "myResource", pass: false},
+		{matches: "Exact resource component match with resource string, with no delimiter", value: "(^myResource$)", resource: "notMyResource", pass: false},
+		{matches: "Exact resource component match with resource string, with no delimiter", value: "(^myResource$)", resource: "MyResourceNot", pass: false},
+		{matches: "Exact resource component match with resource string, with no delimiter", value: "(^myResource$)", resource: "myResource", pass: true},
```

Comment:
@arekkas I made the delimiter optional, so one could choose to use that or go full-blown regex as shown here.
This could make sense if resource strings are not composed of delimited values...

Reply:
That makes sense! I'm giving it one last pass and then, I think, we're good to go :)